src/Compilers/VisualBasic/Portable/Semantics/Conversions.vb

```diff
@@ -1261,12 +1263,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Return Nothing 'ConversionKind.NoConversion
                 End If
 
+                involvesNarrowingFromNumericConstant = involvesNarrowingFromNumericConstant Or elementConversion
+
                 If IsNarrowingConversion(elementConversion) Then
+                    allNarrowingIsFromNumericConstant = allNarrowingIsFromNumericConstant And elementConversion
                     result = narrowingConversion
                 End If
             Next
 
-            Return result
+            Return result Or
+                   (involvesNarrowingFromNumericConstant And allNarrowingIsFromNumericConstant And ConversionKind.InvolvesNarrowingFromNumericConstant)
```

Comment:
If allNarrowingIsFromNumericConstant is "true", then involvesNarrowingFromNumericConstant is not necessary "true". That is because we may never enter the "narrowing" if.